test/libsolidity/syntaxTests/functionTypes/warn_function_return_parameters_no_data_location.sol

```diff
@@ -0,0 +1,5 @@
+library C {
+    function f() private pure returns(uint[]) {}
+}
+// ----
+// Warning: (50-56): Parameter is declared as memory. Use an explicit "memory" keyword to silence this warning.
```

Comment:
Please add a test that shows that this warning can actually be removed by adding `memory`.

Please also add testst for external and internal functions and functions using storage parameters, both in contracts and in libraries.

Reply:
I have added these tests for all types of function visibility as well as for contracts and libraries. I also added tests showing the warning is removed by adding a storage location.